integration/src/main/java/scripts/testscripts/ResourceLineage.java

```diff
@@ -106,20 +148,167 @@ protected void doUserJourney(
+  }
+
+  private void cloneControlledBqDataset() throws ApiException, InterruptedException {
+    // Clone controlled bq dataset 1 into second workspace
+    var cloneRequest =
+        new CloneControlledGcpBigQueryDatasetRequest()
+            .name("clonedControlledDataset2")
+            .jobControl(new JobControl().id(UUID.randomUUID().toString()))
+            .destinationDatasetName("cloned_controlled_dataset2")
+            .cloningInstructions(CloningInstructionsEnum.DEFINITION)
+            .destinationWorkspaceId(workspaceId2);
+    CloneControlledGcpBigQueryDatasetResult resource2CloneResult =
+        controlledGcpResourceApi.cloneBigQueryDataset(
+            cloneRequest, getWorkspaceId(), controlledBqDatasetWorkspace1ResourceId);
+    resource2CloneResult =
+        ClientTestUtils.pollWhileRunning(
+            resource2CloneResult,
+            () ->
+                controlledGcpResourceApi.getCloneBigQueryDatasetResult(
+                    getWorkspaceId(), cloneRequest.getJobControl().getId()),
+            CloneControlledGcpBigQueryDatasetResult::getJobReport,
+            Duration.ofSeconds(5));
+    assertJobSuccess(
+        "clone controlled BigQuery dataset 2",
+        resource2CloneResult.getJobReport(),
+        resource2CloneResult.getErrorReport());
+    var cloneControlledBqDataset2ResourceId =
+        resource2CloneResult.getDataset().getDataset().getMetadata().getResourceId();
+    expectedControlledBqDatasetLineage.add(
+        new ResourceLineageEntry()
+            .sourceResourceId(controlledBqDatasetWorkspace1ResourceId)
+            .sourceWorkspaceId(getWorkspaceId()));
+    assertEquals(
+        expectedControlledBqDatasetLineage,
+        resource2CloneResult.getDataset().getDataset().getMetadata().getResourceLineage());
 
+    // Clone resource 2 to resource 3, still in second workspace.
+    var cloneRequest2 =
+        new CloneControlledGcpBigQueryDatasetRequest()
+            .cloningInstructions(CloningInstructionsEnum.DEFINITION)
+            .destinationWorkspaceId(workspaceId2)
+            .name("clonedControlledDataset3")
+            // Specify the dataset name because there is already a cloned dataset in workspace 2.
+            .destinationDatasetName("cloned_controlled_dataset3")
+            .jobControl(new JobControl().id(UUID.randomUUID().toString()));
+    CloneControlledGcpBigQueryDatasetResult resource3CloneResult =
+        controlledGcpResourceApi.cloneBigQueryDataset(
+            cloneRequest2, workspaceId2, cloneControlledBqDataset2ResourceId);
+    resource3CloneResult =
+        ClientTestUtils.pollWhileRunning(
+            resource3CloneResult,
+            () ->
+                controlledGcpResourceApi.getCloneBigQueryDatasetResult(
+                    workspaceId2, cloneRequest2.getJobControl().getId()),
+            CloneControlledGcpBigQueryDatasetResult::getJobReport,
+            Duration.ofSeconds(5));
     // Assert resource lineage on resource 3.
```

Comment:
nit newline before 218, makes test easier to scan. Same for cloneControlledGcsBucket